tests/test_cases/test_axi4/test_axi4.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env python
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Test to demonstrate functionality of the AXI4 master interface"""
+
+from random import randint, randrange
+
+import cocotb
+from cocotb.clock import Clock
+from cocotb.drivers.amba import (AXIBurst, AXI4LiteMaster, AXI4Master,
+                                 AXIProtocolError, AXIReadBurstLengthMismatch,
+                                 AXIxRESP)
+from cocotb.regression import TestFactory
+from cocotb.result import TestFailure
+from cocotb.triggers import ClockCycles, Combine, Join, RisingEdge
+
+
+CLK_PERIOD = (10, "ns")
+AXI_PREFIX = "S_AXI"
+
+
+def get_parameters(dut):
+    address_width = dut.ADDR_WIDTH.value.integer // 8
+    data_width = dut.DATA_WIDTH.value.integer // 8
+    ram_first = dut.RAM_BASE_ADDRESS.value.integer
+    ram_last = ram_first + 2**dut.RAM_WIDTH.value.integer
```

Comment:
From this alone, I thought you had an off-by-one error, but since you use the value in ``randrange`` (https://docs.python.org/3/library/random.html#random.randrange), the ``ram_last`` value itself is not included.
It might be clearer to use the terminology of ``(rand)range``and call these values ``ram_addr_start`` and ``ram_addr_stop``.